File: 프로그래머스/2/42888. 오픈채팅방/오픈채팅방.py
def solution(record):
    answer = []
    uid = {}
    
    for r in record:
        cmd = r.split(" ") 
        if cmd[0] != "Leave": uid[cmd[1]] = cmd[2]
        
    for r in record:
        cmd = r.split(" ")
        if cmd[0] == "Enter": answer.append("%s님이 들어왔습니다." % uid[cmd[1]])
        elif cmd[0] == "Change": pass
        else: answer.append("%s님이 나갔습니다." % uid[cmd[1]])
    
    return answer

File: 프로그래머스/2/42888. 오픈채팅방/오픈채팅방.js
function solution(record) {
    let answer = []
    const user = new Map()

    record.forEach(v => {
        let [s, uid, nick] = v.split(' ')
        if (s === 'Enter' || s === 'Change') user.set(uid, nick)
    })

    record.forEach(v => {
        let [s, uid] = v.split(' ')
        if (s === 'Enter') answer.push(`${user.get(uid)}님이 들어왔습니다.`)
        if (s === 'Leave') answer.push(`${user.get(uid)}님이 나갔습니다.`)
    })

    return answer
}